packages/assets-controllers/src/TokenBalancesController.test.ts

```diff
@@ -2349,4 +2352,1428 @@ describe('TokenBalancesController', () => {
+      });
+
+      expect(controller).toBeDefined();
+      // Verify that AccountsAPI fetcher is not created when external services are disabled
+      expect(controller.state.tokenBalances).toStrictEqual({});
+    });
+
+    it('should use default allowExternalServices when not provided', () => {
+      // Test line 197: default allowExternalServices = () => true
+      const { controller } = setupController({
+        config: {
+          useAccountsAPI: true,
+          // allowExternalServices not provided - should use default
+        },
+      });
+
+      expect(controller).toBeDefined();
+      expect(controller.state.tokenBalances).toStrictEqual({});
+    });
+
+    it('should handle inactive controller during polling', async () => {
+      const chainId = '0x1';
+      const { controller } = setupController({
+        config: { interval: 100 }, // Short interval to trigger polling quickly
+      });
+
+      // Use fake timers to control polling intervals
+      jest.useFakeTimers();
+
+      // Mock _executePoll to track calls
+      const executePollSpy = jest.spyOn(controller, '_executePoll');
+
+      // Start polling to set up the timer
+      controller.startPolling({ chainIds: [chainId] });
+
+      // Allow initial polling to complete
+      await Promise.resolve();
+      jest.runOnlyPendingTimers();
+      await Promise.resolve();
+
+      // Clear spy calls from setup
+      executePollSpy.mockClear();
+
+      // Stop polling - this makes controller inactive (#isControllerPollingActive = false)
+      controller.stopAllPolling();
+
+      // Fast forward time to trigger the next scheduled poll interval
+      // This should hit line 335 (early return when !#isControllerPollingActive)
+      jest.advanceTimersByTime(150);
+      await Promise.resolve();
+
+      // The scheduled poll should have been prevented by the inactive check (line 335)
+      expect(executePollSpy).not.toHaveBeenCalled();
+      expect(controller).toBeDefined();
+
+      jest.useRealTimers();
+      executePollSpy.mockRestore();
+    });
+
+    it('should handle polling errors with console.warn', async () => {
+      const chainId = '0x1';
+      const consoleWarnSpy = jest
+        .spyOn(console, 'warn')
+        .mockImplementation(() => {
+          return undefined; // Suppress console output during tests
+        });
+
+      const { controller } = setupController({
+        config: { interval: 100 },
+      });
+
+      // Mock _executePoll to throw errors - this will trigger lines 340-343 error handling
+      jest
+        .spyOn(controller, '_executePoll')
+        .mockRejectedValue(new Error('Test polling error'));
+
+      // Use fake timers
+      jest.useFakeTimers();
+
+      // Start polling - this triggers immediate polling and error handling
+      controller.startPolling({ chainIds: [chainId] });
+
+      // Allow immediate polling error to be caught (lines 340-343)
+      await Promise.resolve();
+      await Promise.resolve();
```

Comment:
This is a little confusing, is it for next tick?

Can we replace these with `await flushPromises()`?